src/multiversioning.zig

```diff
@@ -1402,37 +1412,48 @@ pub fn self_exe_path(allocator: std.mem.Allocator) ![:0]const u8 {
 
     const native_self_exe_path = try std.fs.selfExePath(&buf);
 
-    if (builtin.target.os.tag == .linux and std.mem.eql(
-        u8,
-        native_self_exe_path,
-        "/memfd:" ++ multiversion_uuid ++ " (deleted)",
-    )) {
-        // Technically, "/memfd:tigerbeetle-multiversion-... (deleted)" is a valid path at which you
-        // could place your binary - please don't!
+    // Technically, "/memfd:tigerbeetle-multiversion-... (deleted)" is a valid path at which you
+    // could place your binary - please don't!
+    if (builtin.target.os.tag == .linux and
+        std.mem.eql(u8, native_self_exe_path, "/memfd:" ++ multiversion_uuid ++ " (deleted)"))
+    {
+        comptime assert(builtin.target.os.tag == .linux);
         assert(std.fs.cwd().statFile(native_self_exe_path) catch null == null);
 
         // Running from a memfd already; the real path is argv[0].
         const path = try allocator.dupeZ(u8, std.mem.span(os.argv[0]));
         assert(std.fs.path.isAbsolute(path));
 
         return path;
-    } else if (std.mem.indexOf(u8, native_self_exe_path, multiversion_uuid) != null) {
-        assert(builtin.target.os.tag == .windows or builtin.target.os.tag == .macos);
-        // Similar to above, you _could_ call your binary "tigerbeetle-multiversion-...". This can't
-        // be checked with an assert unfortunately.
+    }
+
+    if (builtin.target.os.tag == .macos and
+        std.mem.indexOf(u8, native_self_exe_path, multiversion_uuid) != null)
+    {
+        comptime assert(builtin.target.os.tag == .macos);
 
         // Running from a temp path already; the real path is argv[0].
-        var arg_iterator = try std.process.argsWithAllocator(allocator);
-        defer arg_iterator.deinit();
+        const path = try allocator.dupeZ(u8, std.mem.span(os.argv[0]));
+        assert(std.fs.path.isAbsolute(path));
+
+        return path;
+    }
+
+    if (builtin.target.os.tag == .windows and
+        std.mem.indexOf(u8, native_self_exe_path, multiversion_uuid) != null)
+    {
+        comptime assert(builtin.target.os.tag == .windows);
+        // Windows make it error-prone to set argv[0], so the path is passed via env.
```

Comment:
I suppose we can replicate `argvToCommandLineWindows` from `Child.zig`, but passing via env feels safer!